src/libertem/io/dataset/__init__.py

```diff
@@ -25,12 +25,13 @@ def _auto_load(path, executor):
             "please specify the `path` kwarg to allow auto detection"
         )
 
-    params = detect(path, executor=executor)
-    filetype_detected = params.pop('type', None)
+    detected_params = detect(path, executor=executor)
+    filetype_detected = detected_params.pop('type', None)
```

Comment:
`type` no longer needs to be `pop`ed off - just use `get` instead.

Reply:
Fixed!